chore: add staging workflow

pchandra19 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->


## Description
<!--- Describe your changes in detail -->
Added staging workflow so that we can stage changes and test them prior to release.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
The staging workflows will act as a safe, controlled pre-production validation layer which ensures:
- Everything builds
- Everything passes checks
- Charts are valid
- Images exist
- Workflows run correctly
- No surprises during release

## How Has This Been Tested?
This has been tested in a fork with a simulated environment.

## Types of changes
- [x] Breaking change (This change modifies our CI workflows and scripts)

tiagolobocastro commented on .github/workflows/develop-release-chart.yml:
```diff
@@ -11,16 +12,20 @@ jobs:
     steps:
       - uses: actions/checkout@v4
-      - uses: cachix/install-nix-action@v31.3.0
+      - uses: cachix/install-nix-action@v22
```
Why the version regression?

pchandra19 replied:
The current release-chart.yml had this and I guess since I did not modify that part it was kept as is. I will change it.

tiagolobocastro commented on .github/workflows/develop-release-chart.yml:
```diff
@@ -11,16 +12,20 @@ jobs:
         run: |
           branch="${{ github.ref_name }}"
           nix-shell --pure --run "./scripts/helm/publish-chart-yaml.sh --check-chart "$branch"" ./scripts/helm/shell.nix
+          #nix-shell --pure --run "SKIP_GIT=1 ./scripts/helm/generate-readme.sh" ./scripts/helm/shell.nix
```
this is commented out?

pchandra19 replied:
Should not be. I'll fix this.

tiagolobocastro commented on .github/workflows/develop-release-chart.yml:
```diff
@@ -1,8 +1,9 @@
   push:
     branches:
       - develop
+      - 'release/**'
```
hmm won't this publish the release?

pchandra19 replied:
this will publish the develop and release branch charts to gh-pages. This is on push to branch.

tiagolobocastro replied:
hmm no this will push the stable release

pchandra19 replied:
Ah, yes because we update the chart version to actual version (to be released) upon release branch. Hmm.. Let me make the adjustments accordingly.

pchandra19 replied:
I have put a condition to only publish on develop.

tiagolobocastro commented on .github/workflows/release-binaries.yml:
```diff
@@ -18,10 +22,27 @@ jobs:
             target: linux-musl
             arch: x86_64
-          - os: oracle-vm-8cpu-32gb-x86-64
+          - os: ubuntu-latest
```
this was just for testing right?

pchandra19 replied:
yes, I forgot to change while copying the files over. Let me fix that.

tiagolobocastro commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,169 @@
+
+env:
+    TAG: ${{ github.ref_name }}
+    CI: 1
```
```suggestion
  TAG: ${{ github.ref_name }}
  CI: 1
```

pchandra19 replied:
Fixed

tiagolobocastro commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,169 @@
+          export NIX_PATH=nixpkgs=$(jq '.nixpkgs.url' nix/sources.json -r)
+          echo "NIX_PATH=$NIX_PATH" >> $GITHUB_ENV
+          nix-shell --pure --run "echo" ./dependencies/control-plane/utils/dependencies/scripts/staging/shell.nix
+
```
```suggestion
```

pchandra19 replied:
FIxed

tiagolobocastro commented on .github/workflows/staging.yml:
```diff
@@ -0,0 +1,136 @@
+          registry: ghcr.io
+          registry_username: ${{ github.actor }}
+          registry_password: ${{ secrets.GITHUB_TOKEN }}
+          update_dependencies: 'true'
```
Already updated above

```suggestion
          update_dependencies: 'false'
```

pchandra19 replied:
Fixed

openebs-ci committed:
chore(ci): update git submodules

Signed-off-by: GitHub <noreply@github.com>

Commit:
chore(bors): merge pull request #790

790: [CI] Update git submodules r=openebs-ci a=openebs-ci

Automated changes by [create-pull-request](https://github.com/peter-evans/create-pull-request) GitHub action

Co-authored-by: openebs-ci <137901803+openebs-ci@users.noreply.github.com>
